draft-irtf-cfrg-opaque.md

```diff
@@ -1545,10 +1552,13 @@ Input:
 Output:
 - session_key, the shared session secret if, and only if, KE3 is valid, nil otherwise.
 
+Exceptions:
+- HandshakeError, when the handshake fails
+
 Steps:
-1. if ct_equal(ke3.client_mac, state.expected_client_mac):
-2.    Output state.session_key
-3. Output nil
+1. if !ct_equal(ke3.client_mac, state.expected_client_mac):
+2.    raise HandshakeError
+3. Output state.session_key
```

Comment:
This change makes the client and server interfaces different: both either return something, or toss an exception.